### ISSUE TITLE
Possible memory leak leading to recurring OOMKilled

### ISSUE THREAD

**Issue by dkulchinsky** [c4po/harbor_exporter#52]
Hey there @c4po!

We've been testing the exporter in our sandbox environment for few weeks now and it looks like there might be a memory leak causing OOMKilled condition.

Our Harbor setup runs in K8s (GKE), harbor-exporter runs in a separate pod with the following resource allocations:
```
    Limits:
      cpu:     400m
      memory:  512Mi
    Requests:
      cpu:      100m
      memory:   64Mi
```

this should reasonable, especially considering the sandbox deployment has 2 projects and a handful of repos (no replications).

Looking at the memory usage graph it seems to suggest that memory usage grows pretty quickly over time until the limit is reached and the pod is OOMkilled.
![image](https://user-images.githubusercontent.com/61992380/94813743-65b99080-03c6-11eb-80d9-5fb8736bacb3.png)

I didn't try to debug the root cause yet, going to try and enable metric caching and see if that helps, will report back.

If there's any additional info you need, please let me know.

**Comment by Nick-Triller** [comment]
Hi,

I found the leak. For each request, a new HTTP client is created here: https://github.com/c4po/harbor_exporter/blob/master/harbor_exporter.go#L227
This behavior was introduced in v0.5.0

netstat clearly shows that new connections are opened and stay open for each scrape request:
```
watch -n 0.1 "nsenter -t $(docker inspect -f '{{.State.Pid}}' harbor-exporter) --mount --net --pid -n netstat -n  | wc -l"
```

I created a fix and will open a pull request soon.

Best
Nick

**Comment by dkulchinsky** [comment]
good catch @Nick-Triller, happy to test the fix once you have it.

**Comment by Nick-Triller** [comment]
@dkulchinsky Feel free to test the fix now: https://github.com/c4po/harbor_exporter/pull/55

**Comment by c4po** [comment]
thanks @Nick-Triller @dkulchinsky , this has been merged and released as v0.5.7 .
